src/shared/lib/geography/index.js

```diff
@@ -103,6 +103,10 @@ export const isCity = function(location) {
   return location && location.city;
 };
 
+export const getSmallestLocationStr = function(location) {
+  return location.city || location.county || location.state || location.country || null;
```

Comment:
Shouldn't this throw, instead of going to null?  Optionally, if we don't even know the country, perhaps it should return "Earth".

Reply:
Good point. I believe no magic things, so definitely not "Earth", that'd wreak havoc on the datastructure at one point, but throwing is an interesting idea.

@lazd @camjc @qgolsteyn @ryanblock can you imagine a situation where all of location.city || location.county || location.state || location.country are not defined?

Reply:
Throwing is safest. Use an “assert” perhaps.
God I hope we don’t hit that case though, that’d be pretty odd.

Reply:
I've just implemented it with assert, guess what, JHU is an illegal location.

Reply:
The best way to keep JHU a valid "location" might be to add country = "JHU" on that one.

Reply:
The other 2 are:
- /AUS/_shared/get-data-with-tested-negative-applied.js
- /AUS/_shared/get-key.js

This is so strange, why are these picked up as locations? For the _shared ones, we should add "_shared" them to the glob filter, shouldn't we? 

For JHU, we can solve it by adding country = '-' or country = 'JHU'. Or just using null as in the original version.

Reply:
> can you imagine a situation where all of location.city || location.county || location.state || location.country are not defined? That would be a pretty bad scenario, so I think I'm into throwing in that case!